bumble/device.py

```diff
@@ -2804,24 +2804,32 @@ def on_pin_code_request(self, connection):
 
             async def get_pin_code():
                 pin_code = await connection.abort_on(
-                    'disconnection', pairing_config.delegate.get_number()
+                    'disconnection', pairing_config.delegate.get_string(16)
                 )
 
                 if pin_code is not None:
-                    pin_code = bytes(str(pin_code).zfill(6))
-                    await self.host.send_command(
-                        HCI_PIN_Code_Request_Reply_Command(
-                            bd_addr=connection.peer_address,
-                            pin_code_length=len(pin_code),
-                            pin_code=pin_code,
+                    pin_code = bytes(pin_code, encoding='utf8')
+                    pin_code_len = len(pin_code)
+                    if 0 < pin_code_len <= 16:
```

Comment:
Since `max_length` was already supplied to `get_string` I would only assert here to make sure the API contract has been respected